rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1806,4 +1806,36 @@ private void unselectNodes(final HashSet<RMNode> selectedRMNodes) {
         }
     }
 
+    protected void setNodeSources(HashMap<String, NodeSource> hashMap) {
```

Comment:
Normally we should not compromise encapsulation for the sake of unit testing. But I know it is not possible to mock internally created objects due to the current legacy design :(